keras_cv/models/classification/image_classifier_test.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for ImageClassifier."""
+
+
+import os
+
+import pytest
+import tensorflow as tf
+from absl.testing import parameterized
+from tensorflow import keras
+
+from keras_cv.models.backbones.resnet_v2.resnet_v2_backbone import (
+    ResNet18V2Backbone,
+)
+from keras_cv.models.classification.image_classifier import ImageClassifier
+
+
+class ImageClassifierTest(tf.test.TestCase, parameterized.TestCase):
+    def setUp(self):
+        self.input_batch = tf.ones(shape=(8, 224, 224, 3))
+        self.dataset = tf.data.Dataset.from_tensor_slices(
+            (self.input_batch, tf.one_hot(tf.ones((8,), dtype="int32"), 2))
+        ).batch(4)
+        self.model = ImageClassifier(
```

Comment:
should we move model construction into tests across all cases for consistency?  Right now we build a model in some tests, and use this one for others.  Its a bit confusing this way.

Reply:
Sure, happy to do that